util/src/main/java/tc/oc/pgm/util/nms/NMSHacksNoOp.java

```diff
@@ -0,0 +1,192 @@
+
+    // Some effects can be parsed by getByName, this only has cases for those that cannot
+    switch (strippedKey) {
+      case "slowness":
+        return PotionEffectType.SLOW;
+      case "haste":
+        return PotionEffectType.FAST_DIGGING;
+      case "miningfatigue":
+        return PotionEffectType.SLOW_DIGGING;
+      case "strength":
+        return PotionEffectType.INCREASE_DAMAGE;
+      case "instanthealth":
+        return PotionEffectType.HEAL;
+      case "instantdamage":
+        return PotionEffectType.HARM;
+      case "jumpboost":
+        return PotionEffectType.JUMP;
+      case "nausea":
+        return PotionEffectType.CONFUSION;
+      case "resistance":
+        return PotionEffectType.DAMAGE_RESISTANCE;
+      default:
+        return PotionEffectType.getByName(key);
+    }
+  }
+
+  @Override
+  public Enchantment getEnchantment(String key) {
+    String strippedKey = key.toLowerCase().replace(" ", "").replace("_", "");
+
+    switch (strippedKey) {
+      case "protection":
+        return Enchantment.PROTECTION_ENVIRONMENTAL;
+      case "fireprotection":
+        return Enchantment.PROTECTION_FIRE;
+      case "featherfalling":
+        return Enchantment.PROTECTION_FALL;
+      case "blastprotection":
+        return Enchantment.PROTECTION_EXPLOSIONS;
+      case "projectileprotection":
+        return Enchantment.PROTECTION_PROJECTILE;
+      case "respiration":
+        return Enchantment.OXYGEN;
+      case "aquaaffinity":
+        return Enchantment.WATER_WORKER;
+      case "thorns":
+        return Enchantment.THORNS;
+      case "depthstrider":
+        return Enchantment.DEPTH_STRIDER;
+      case "sharpness":
+        return Enchantment.DAMAGE_ALL;
+      case "smite":
+        return Enchantment.DAMAGE_UNDEAD;
+      case "baneofarthropods":
+        return Enchantment.DAMAGE_ARTHROPODS;
+      case "knockback":
+        return Enchantment.KNOCKBACK;
+      case "fireaspect":
+        return Enchantment.FIRE_ASPECT;
+      case "looting":
+        return Enchantment.LOOT_BONUS_MOBS;
+      case "efficiency":
+        return Enchantment.DIG_SPEED;
+      case "silktouch":
+        return Enchantment.SILK_TOUCH;
+      case "unbreaking":
+        return Enchantment.DURABILITY;
+      case "fortune":
+        return Enchantment.LOOT_BONUS_BLOCKS;
+      case "power":
+        return Enchantment.ARROW_DAMAGE;
+      case "punch":
+        return Enchantment.ARROW_KNOCKBACK;
+      case "flame":
+        return Enchantment.ARROW_FIRE;
+      case "infinity":
+        return Enchantment.ARROW_INFINITE;
+      case "luckofthesea":
+        return Enchantment.LUCK;
+      case "lure":
+        return Enchantment.LURE;
+      default:
+        return Enchantment.getByName(key);
+    }
+  }
+
+  @Override
+  public long getMonotonicTime(World world) {
+    return Instant.now().toEpochMilli() / 50;
+  }
+
+  @Override
+  public int getPing(Player player) {
+    return 0;
```

Comment:
i'd use a more reasonable default like 100ms for this

Reply:
Will do 👍